dd4t-api/src/main/java/org/dd4t/core/util/TCMURI.java

```diff
@@ -102,6 +114,16 @@ protected void load(String uriString) throws ParseException {
         }
     }
 
+    private static boolean startsWithNamespace(String uriString) {
+        for (Namespace namespace : Namespace.values()) {
+            if (uriString.startsWith(namespace.getValue())) {
```

Comment:
When you introduce a static map in enum, you'll have an easy and elegant .containsKey()